scripts/in_container/run_resource_check.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import psutil
+from rich.console import Console
+
+console = Console(force_terminal=True, color_system="standard", width=180)
+
+
+def get_size(kilobytes):
+    """
+    Convert kilobytes into gigabytes
+    1 Gigabytes = 1,048,576 Kb
+    """
+    factor = 1000000000
+    value_gb = kilobytes // factor
+    return value_gb
+
+
+def resoure_check():
+    """
+    Use gsutil to get resources in kylobytes: memory, cpus and disk
+    """
+    resources = {}
+    print("\nChecking resources.\n")
+
+    # Memory available
+    svmem = psutil.virtual_memory()
+    mem_available = get_size(svmem.total)
+    resources.setdefault('Memory', []).append(mem_available)
```

Comment:
```suggestion
    resources= {
        'Memory': [mem_available, MAX_MEMORY]
        'Cpus' : ....
```

This would be far simpler and shorter (no need to use `setdefault` here

Reply:
Actually it would be better to use a dataClass instead of array:

```
@dataclass
class Resource:
   current: float
   max: float
```

and then

```
resources: Dict[str, Resource] = {
     'Memory': Resource(cuurent=mem_available, maxMAX_MEMORY),
     'Cpus': ...
}
```

Then you could refer to it via:

```
available.current, available.max
```
rather than

```
available[0], available[1]
```

Reply:
I changed for dataclass @potiuk, I tried to give the variables an appropriate name.